motr/io_req.c

```diff
@@ -1314,8 +1314,10 @@ static int ioreq_application_data_copy(struct m0_op_io *ioo,
 		/* verify the checksum during data read.
 		 * skip checksum verification during degraded I/O
 		 */
-		if (ioreq_sm_state(ioo) != IRS_DEGRADED_READING &&
-		    ioo->ioo_attr.ov_vec.v_nr &&
+		if (ioo->ioo_dgmode_io_sent == false &&
+			total_count % usz == 0 &&
```

Comment:
Is this needed, if we remove this any failure?

Reply:
yes, there is failure in **motr_utils_st**.